scripts/travis/install-deps.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# install dependencies
+# (this script must be run as root)
+
+BASEDIR=$(dirname $0)
+source $BASEDIR/defaults.sh
+
+apt-get -y update
+apt-get install -y --no-install-recommends \
+  build-essential \
+  libboost-filesystem-dev \
+  libboost-python-dev \
+  libboost-system-dev \
+  libboost-thread-dev \
+  libgflags-dev \
+  libgoogle-glog-dev \
+  libhdf5-serial-dev \
+  libopenblas-dev \
+  python-virtualenv \
+  wget
+
+if $WITH_CMAKE ; then
+  apt-get install -y --no-install-recommends cmake
+fi
+
+# build Protobuf3 since it's needed for Python3
+PROTOBUF3_DIR=~/protobuf3-build
```

Comment:
Possibly. I'm looking into how the cache works, and I figured it might thrash between uploading an empty cache directory from one build and then uploading a full cache directory from another. So I thought I'd just set it so that each build would build from source the first time, and then none would.

As I said, I'm still looking into whether that's true. It's tricky to test the cache system.